cpp/src/trajectory/subset_trajectories.cu

```diff
@@ -120,37 +116,41 @@ struct subset_functor {
 
 namespace cuspatial {
 
-uint32_t subset_trajectory_id(const gdf_column& id,
-                              const gdf_column& in_x, const gdf_column& in_y,
-                              const gdf_column& in_id,
-                              const gdf_column& in_timestamp,
-                              gdf_column& out_x, gdf_column& out_y,
-                              gdf_column& out_id, gdf_column& out_timestamp)
-{       
-    struct timeval t0,t1;
-    gettimeofday(&t0, nullptr);
-    
+gdf_size_type subset_trajectory_id(const gdf_column& id,
+                                   const gdf_column& in_x,
+                                   const gdf_column& in_y,
+                                   const gdf_column& in_id,
+                                   const gdf_column& in_timestamp,
+                                   gdf_column& out_x,
+                                   gdf_column& out_y,
+                                   gdf_column& out_id,
+                                   gdf_column& out_timestamp)
+{
     CUDF_EXPECTS(in_x.data != nullptr && in_x.data != nullptr &&
                  in_id.data != nullptr && in_timestamp.data != nullptr,
-                 "x/y/in_id/in_ts data cannot be null");
+                 "Null input data");
     CUDF_EXPECTS(in_x.size == in_y.size && in_x.size == in_id.size &&
-                 in_x.size == in_timestamp.size ,
-                 "x/y/in_id/timestamp must have equal size");
-    
-    // future versions might allow x/y/in_id/timestamp to have null_count > 0,
-    // which might be useful for taking query results as inputs 
+                 in_x.size == in_timestamp.size,
+                 "Data size mismatch");
+    CUDF_EXPECTS(in_id.dtype == GDF_INT8 || in_id.dtype == GDF_INT16 ||
```

Comment:
In `subset_functor`, `in_id` is always being casted to `int32_t`, which isn't valid if the type is anything other than `GDF_INT32`.

Reply:
Fixed.